File: SirenView.java
import javafx.animation.Animation;
import javafx.animation.KeyFrame;
import javafx.animation.Timeline;
import javafx.scene.paint.Color;
import javafx.scene.shape.Polygon;
import javafx.util.Duration;

public class SirenView extends Polygon {
    public SirenView() {
        getPoints().addAll(0d,30d,
                0d,50d,
                40d, 80d,
                40d, 0d);
        setFill(Color.web("4FEC4C"));
        setStroke(Color.GREEN);
        timeline = new Timeline(new KeyFrame(Duration.millis(250),e -> startBlinking()));
        timeline.setCycleCount(Animation.INDEFINITE);
    }
    public void startBlinking(){
        timeline.play();
        setFill(Color.web("F03B3B"));
        setStroke(Color.RED);
    }
    public void stopBlinking() {
        timeline.stop();
        setFill(Color.web("4FEC4C"));
        setStroke(Color.GREEN);
    }
    private final Timeline timeline;
}

File: sirenview.cpp
#include "sirenview.h"
#include <QBrush>
#include <iostream>

SirenView::SirenView(){
    makeSirenView();
}
void SirenView::makeSirenView(){
    siren=new QGraphicsPolygonItem(this);
    QPolygonF p;
    p.append(QPoint(0,30));
    p.append(QPoint(0,50));
    p.append(QPoint(40,80));
    p.append(QPoint(40,0));
    siren->setPolygon(p);
    siren->setBrush(Qt::green);
    addToGroup(siren);
}
void SirenView::startBlinking(){
    siren->setBrush(Qt::red);
}
void SirenView::stopBlinking(){
    siren->setBrush(Qt::green);
}
SirenView::~SirenView(){
    delete siren;
}
